src/connectors/youtube.js

```diff
@@ -174,25 +177,46 @@ function getVideoCategory(videoId) {
 	if (videoId === null) {
 		return null;
 	}
-	if (!categoryCache.has(videoId)) {
-		const url = `https://www.googleapis.com/youtube/v3/videos?part=snippet&id=${videoId}&key=${YT_API_KEY}`;
-		fetch(url).then((response) => {
+	if (categoryCache.has(videoId)) {
+		fetchCategoryId(videoId).then((category) => {
+			if (category === null) {
+				throw Error(`Failed to resolve category for ${videoId}`);
+			}
+			categoryCache.set(videoId, category);
+		}).catch((err) => {
+			console.log(err);
+		});
+	}
+
+	return categoryCache.get(videoId);
+}
+
+async function fetchCategoryId(videoId) {
+	let found = false;
+	for (let i = 0; i < YT_API_KEYS.length; i++) {
+		let key = YT_API_KEYS[i];
+		const url = `https://www.googleapis.com/youtube/v3/videos?part=snippet&id=${videoId}&key=${key}`;
+		try {
+			const response = await fetch(url);
 			if (!response.ok) {
-				throw new Error('Invalid response');
+				throw Error(response.statusText);
 			}
-			return response.json();
-		}).then((data) => {
+
+			const data = await response.json();
+
 			let category = data.items[0].snippet.categoryId;
 			if (typeof category === 'string') {
-				categoryCache.set(videoId, category);
+				return category;
 			}
-		}).catch((err) => {
-			console.log(`Unable to get category for ${videoId}: ${err.message}`);
-		});
-		return null;
+		}
+		catch (err) {
+			console.log(`Failed fetching category for ${videoId} due to ${err}`);
+		}
```

Comment:
It's ok, but almost all code is in this format:
```js
if (condition) {
    statement
} else {
    statement
}
```
The same thing for `try...catch` blocks.

Reply:
Ahh formatting ;)

Reply:
WOuldn't this be it? https://eslint.org/docs/rules/brace-style 1tbs which we already use?

Reply:
It's already set up for 1TBS, but does not work for `try...catch` statements.

https://eslint.org/docs/rules/brace-style